Godeps/_workspace/src/github.com/GoogleCloudPlatform/kubernetes/pkg/runtime/interfaces.go

```diff
@@ -52,6 +95,20 @@ type ObjectCreater interface {
 	New(version, kind string) (out Object, err error)
 }
 
+// ObjectDecoder is a convenience interface for identifying serialized versions of objects
+// and transforming them into Objects. It intentionally overlaps with ObjectTyper and
+// Decoder for use in decode only paths.
+type ObjectDecoder interface {
```

Comment:
----- Original Message -----
> > @@ -52,6 +95,20 @@ type ObjectCreater interface {
> >  	New(version, kind string) (out Object, err error)
> >  }
> >  
> > +// ObjectDecoder is a convenience interface for identifying serialized
> > versions of objects
> > +// and transforming them into Objects. It intentionally overlaps with
> > ObjectTyper and
> > +// Decoder for use in decode only paths.
> > +type ObjectDecoder interface {
> 
> I think I'm seeing it now.  `Object*` types are intended to hold all possible
> `Codecs` for a particular `Kind`?  If that's the case, I'd like to see them
> in a different file and some sort of comment to make that more obvious.
> 

Codecs are distinct from scheme behavior.  A codec is a round trip through a particular serialization.  The remaining interfaces are for dealing with schemes.  interfaces.go is usually where we put interfaces - I think long term we'll put it in another package (runtime/codec) and that will address your concern.